refactor (#330)
KxSystems/docs: docs/wp/compress/index.md

Early versions of kdb+ achieved compression via file systems with inbuilt compression, such as ZFS. V2.7 introduced built-in OS-agnostic compression, which allowed on-disk data to be converted to compressed format using a range of algorithms and compression levels. This was expanded (V2.8), with the ability to stream in-memory data directly to compressed format on disk. ZFS compression is still useful for some kdb+ applications, as it keeps cached data available for multiple processes. However, this paper will focus on the inbuilt data-compression options provided by kdb+, available on all supported architectures.

> Each system will have its own characteristics which determine the appropriate compression configurations to use.

Reviewer: ```suggestion
Each system has its own characteristics, which determine the appropriate compression configurations to use.
```
